### DISCUSSION TITLE
Unable to get Bundle Starter App working.

### DISCUSSION THREAD

**Discussion by devinatbryt** [Shopify/function-examples#197] [Feedback: Cart Transform API]
Hello, I've gone through the tutorial here many times and cannot seem to get this basic example working in dev mode as expected. I've defined the metafields, copy and pasted their names and made sure not to update the keys as referenced in the screenshots below:

![image](https://user-images.githubusercontent.com/101842591/224869794-a67ff3c5-b3a3-4f49-9288-f03d6ccb18cd.png)
![image](https://user-images.githubusercontent.com/101842591/224869891-c0947eea-3128-4a9e-ad09-932050f2da87.png)
![image](https://user-images.githubusercontent.com/101842591/224869972-4465cade-e073-4975-8994-67091c0adf0d.png)
![image](https://user-images.githubusercontent.com/101842591/224870037-5cd13bcc-fd13-4da5-aa69-86f3e38a2ddf.png)

Furthermore, I've also made sure to create the bundle parent product and then tie the associated bundle items via metafields to the bundle as referenced here:

![image](https://user-images.githubusercontent.com/101842591/224870301-80501525-5119-45c8-b7da-5f0dba43764d.png)
![image](https://user-images.githubusercontent.com/101842591/224870447-4fa2ba6c-9d3e-447e-b09b-4a31c471d6a6.png)

Here is the component parents json obj btw:
```json
[{
	"id": "gid://shopify/ProductVariant/44640619135279",
	"component_reference": {
		"value": ["gid://shopify/ProductVariant/44640893632815"]
	},
	"component_quantities": {
		"value": [1]
	},
	"price_adjustment": {
		"value": 1
	}
}]
```

I've also ensure to have the app and extension published. But even while running: "npm run dev" or "pnpm dev" the items at the checkout phase do not bundle. Is there perhaps a step that I am missing? Thank you for looking into this :) !

![image](https://user-images.githubusercontent.com/101842591/224871362-c3f213b0-6b9b-49ef-9f4c-76b53b6b8986.png)

**Comment by rdavid1099** [comment]
Hello @devinatbryt. Thank you for reaching out. We've been making a lot changes and improvements to the `cart_transform` function and appreciate all the feedback during this dev preview window.

Looking over your screenshots, it looks like all of the `metafields` are defined correctly, and I don't see anything wrong with your configuration. I have a couple questions to help narrow down where the problem may be occurring.

1. Is the app installed on your store with all of the [requirements](https://shopify.dev/docs/apps/bundles/add-a-custom-bundle#requirements) satisfied?
2. After installing the app, can you confirm that the app is [registered to the shop](https://github.com/Shopify/function-examples/tree/main/sample-apps/bundles-cart-transform#extension-deployment)? The sample app should handle that automatically. If you check your application's logs, you should see if that request was successful.
3. If you are still running into issues, could you please try to install our [bundles reference app](https://shopify.dev/docs/apps/bundles/create#step-1-install-the-shopify-bundles-sample-app) and see if that works on your shop? This reference app is essentially the exact same code as the sample app, but it's hosted by Shopify during the dev preview window of Bundles. If the reference app works on your store and the sample app doesn't then we can narrow down the problem a bit.

Thanks again for reaching out.

**Reply by devinatbryt** [reply]
**I sincerely apologize for the very late reply here.**
1. Yes I have the app installed on my dev store with all of the aforementioned requirements satisfied. I made sure to thoroughly, copy and paste xD, the provided code and follow the steps provided in the tutorial.
2. I'm trying to test the extension in development, so I haven't deployed the extension yet. Is this a requirement in order to test that the extension works?
3. I am still running into issues, however, I did install the Shopify bundle test app and that works as intended.

Thank you for your patience in this matter and look forward to finding the solution.
